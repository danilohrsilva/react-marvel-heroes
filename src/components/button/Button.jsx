import React from 'react';

const Button = props => {
  const { className, text, icon, disabled } = props;
  return (
    <button className={className} disabled={disabled} >
      { icon ? icon : null }
      { text ? text : null }
    </button>
  );
};

export default Button;
