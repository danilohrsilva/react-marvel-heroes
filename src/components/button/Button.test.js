import React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from './Button';

describe('Given the <Button /> component with all params', () => {

  const mockClick = jest.fn();
  const component = TestRenderer.create(
    <Button className="button-class"
            text="Click Me"
            icon="button-icon"
            disabled={false}
            onClick={mockClick} />
  );

  describe('When it renders, then', () => {

    it('Should match the snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    })

  });


});