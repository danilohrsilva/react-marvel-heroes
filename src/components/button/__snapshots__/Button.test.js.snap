// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the <Button /> component with all params When it renders, then Should match the snapshot 1`] = `
<button
  className="button-class"
  disabled={false}
  onClick={[MockFunction]}
>
  <image
    className="button-class-icon"
    src="button-icon"
  />
  Click Me
</button>
`;
